streams/src/test/java/org/apache/kafka/streams/state/StateSerdesTest.java

```diff
@@ -88,20 +93,47 @@ public void shouldThrowIfValueClassIsNull() {
         new StateSerdes<>("anyName", Serdes.ByteArray(), null);
     }
 
-    @Test(expected = StreamsException.class)
+    @Test
     public void shouldThrowIfIncompatibleSerdeForValue() throws ClassNotFoundException {
         final Class myClass = Class.forName("java.lang.String");
         final StateSerdes<Object, Object> stateSerdes = new StateSerdes<Object, Object>("anyName", Serdes.serdeFrom(myClass), Serdes.serdeFrom(myClass));
         final Integer myInt = 123;
-        stateSerdes.rawValue(myInt);
+        final Exception e = assertThrows(StreamsException.class, () -> stateSerdes.rawValue(myInt));
+        assertThat(
+            e.getMessage(),
+            equalTo(
+                "A serializer (value: org.apache.kafka.common.serialization.StringSerializer) " +
```

Comment:
Could we reuse the error message or the entire template?

Reply:
We could reuse for this and the second "value" test, but not for the "key" test. Don't see big value is sharing though.